api/src/context.rs

```diff
@@ -306,20 +290,39 @@ impl Context {
         Ok((latest_ledger_info, requested_ledger_version))
     }
 
-    pub fn get_latest_internal_indexer_ledger_version_and_main_db_info<E: StdApiError>(
+    pub fn get_latest_internal_indexer_ledger_version_and_ledger_info<
+        E: ServiceUnavailableError,
+    >(
         &self,
     ) -> Result<(LedgerInfo, Version), E> {
         if let Some(indexer_reader) = self.indexer_reader.as_ref() {
             if let Some(latest_version) = indexer_reader
                 .get_latest_internal_indexer_ledger_version()
-                .map_err(|err| E::internal_with_code_no_info(err, AptosErrorCode::InternalError))?
+                .map_err(|err| {
+                    E::service_unavailable_with_code_no_info(err, AptosErrorCode::InternalError)
+                })?
             {
-                let latest_ledger_info = self.get_latest_ledger_info()?;
+                let (_, _, new_block_event) = self
+                    .db
+                    .get_block_info_by_version(latest_version)
+                    .map_err(|_| {
+                        E::service_unavailable_with_code_no_info(
+                            "Failed to get block",
+                            AptosErrorCode::InternalError,
+                        )
+                    })?;
+
+                let mut latest_ledger_info = self.get_latest_storage_ledger_info()?;
+                latest_ledger_info.ledger_version = U64::from(latest_version);
+                latest_ledger_info.epoch = U64::from(new_block_event.epoch());
+
+                latest_ledger_info.block_height = U64::from(new_block_event.height());
+                latest_ledger_info.ledger_timestamp = U64::from(new_block_event.proposed_time());
```

Comment:
most of the work `self.get_latest_storage_ledger_info()` has done is wasted here (getting the latest ledger info, getting the latest NewBlockEvent), and getting a struct then overwriting many fields looks ugly.

Shall we extract out the logic of getting the oldest version and height and call that here?